packages/eslint-plugin/src/rules/ban-types.ts

```diff
@@ -41,6 +41,51 @@ function getCustomMessage(
   return '';
 }
 
+function reportBannedType(
```

Comment:
if you keep this function within the rule `create`, then you don't need to add a new typed `context` arg as you can consume it from the parent closure

Reply:
When I originally wrote this I wasn't sure whether I should prioritize optimal readability or prioritize not having a closure in each `create` call. For some reason, when I wrote this, I was thinking `create` might be called with each RuleListener fire. Now I realize that doesn't make sense.